tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TaskMapper.java

```diff
@@ -300,4 +302,21 @@ private TesResources createTesResources(PipelineRun run) {
                 .logs(tesExecutorLogList)
                 .build());
     }
+
+    private InstanceType getInstanceType(PipelineRun run) {
+        Long regionId = run.getInstance().getCloudRegionId();
+        Boolean spot = run.getInstance().getSpot();
+        Tool pipeLineTool = loadToolByTesImage(run.getDockerImage());
+        String nodeType = run.getInstance().getNodeType();
+        AllowedInstanceAndPriceTypes allowedInstanceAndPriceTypes =
+                cloudPipelineAPIClient.loadAllowedInstanceAndPriceTypes(pipeLineTool.getId(), regionId, spot);
+        return Stream.of(allowedInstanceAndPriceTypes)
+                .map(instance -> instance.getAllowedInstanceTypes())
+                .flatMap(instanceType -> instanceType
+                        .stream()
+                        .filter(instance -> instance
+                                .getName()
+                                .equalsIgnoreCase(nodeType)))
+                .findFirst().get();
```

Comment:
bad practice to use get() for Optional without any check that value actually exists. In this case please use orElseThrow